compiler/rustc_trait_selection/src/solve/project_goals.rs

```diff
@@ -453,7 +456,8 @@ impl<'tcx> assembly::GoalKind<'tcx> for ProjectionPredicate<'tcx> {
                                 tcx,
                                 ty::Binder::dummy(goal.predicate.with_self_ty(tcx, self_ty)),
                             );
-                            return ecx.evaluate_all_and_make_canonical_response(vec![new_goal]);
+                            let (_, certainty) = ecx.evaluate_goal(new_goal)?;
```

Comment:
Why do we throw away `changed` here, but `evaluate_all` runs this in a loop?

Reply:
because `evaluate_all` has multiple goals which can influence each other.

added a second commit which checks that the result of `evaluate_goal` is stable and we don't get any additional information by rerunning a goal after applying its substitutions.

Will have to change the `is_identity` check to ignore regions once they are implemented but that doesn't yet matter.